src/hotspot/os/posix/os_posix.cpp

```diff
@@ -639,6 +639,21 @@ bool os::has_allocatable_memory_limit(size_t* limit) {
 #endif
 }
 
+void* os::get_default_process_handle() {
+#ifdef __APPLE__
+  // MacOS X needs to use RTLD_FIRST instead of RTLD_LAZY
+  // to avoid finding unexpected symbols on second (or later)
+  // loads of a library.
```

Comment:
I find this a puzzling comment. RTLD_FIRST is not used instead of RTLD_LAZY it is used with it. RTLD_FIRST simply limits how dlsym will perform a search using the handle that dlopen returns. Second it is trying to protect against something that will happen on other Posix systems anyway, so can't really be a bug. So I have to question whether we really need to special case this code for macOS?

Reply:
hi David, thanks for your comment and questions. Maybe some Mac expert could comment why the special handling was needed ? Indeed it would be good to remove the special handling and have the same code on all Posix platforms.

Best regards, Matthias